### ISSUE TITLE
Refactor producer/consumer API

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11409]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#390|
|Type|enhancement|
|Created|2003-11-08 10:37:54Z|

<details><summary>Searchable metadata</summary>

```
trac-id__390 390
type__enhancement enhancement
reporter__itamarst itamarst
priority__low low
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__ 
time__1068287874000000 1068287874000000
changetime__1469181085161947 1469181085161947
version__ 
owner__ 
cc__glyph cc__itamarst cc__PenguinOfDoom
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
Instead of producers, you have transport.registerForWritable(callback) and
.registerForFull(callback), the callbacks getting called when the buffers are
empty and transport is writable, and the other when the buffers are full.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
Guh I suck. Duplicate of issue 79.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|GitHub commented|
|-|-|

In changeset da03223ef5309a7184b3e590ca773154ebd3a77c
```
#!CommitTicketReference repository="" revision="da03223ef5309a7184b3e590ca773154ebd3a77c"
Merge pull request #390: Move notice about Python 2.6 from 15.5 to 15.4

Author: glasnt
Reviewer: Lukasa
Fixes: #8651
```
